pkg/quota/api/types.go

```diff
@@ -23,16 +23,27 @@ type ClusterResourceQuota struct {
 
 // ClusterResourceQuotaSpec defines the desired quota restrictions
 type ClusterResourceQuotaSpec struct {
-	// Selector is the label selector used to match projects.  It is not allowed to be empty
-	// and should only select active projects on the scale of dozens (though it can select
+	// Selector is the selector used to match projects.
+	// It should only select active projects on the scale of dozens (though it can select
 	// many more less active projects).  These projects will contend on object creation through
 	// this resource.
-	Selector *unversioned.LabelSelector
+	Selector ClusterResourceQuotaSelector
```

Comment:
@smarterclayton We agreed in principle in person.  You good with the representation?